xds/src/main/java/com/linecorp/armeria/xds/ResourceParser.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2023 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.xds;
+
+import com.google.common.base.MoreObjects;
+import com.google.protobuf.Message;
+
+abstract class ResourceParser<T extends Message> {
```

Comment:
I'm not sure the generic type parameter provides type safety in xDS implementation.
As the input is dynamically determined by `typeUrl` at runtime, the compile time does not seem to work.
https://github.com/line/armeria/pull/5342/files#diff-463524dcc65fabda54bb06752b0e248879c7bc1e5b367d0e8885bfbd23aa785bR209

What do you think of removing `<T extends Message>` and checking the type with `XdsType` only?